app/wizards/placements/add_school_contact_wizard/school_contact_step.rb

```diff
@@ -3,13 +3,14 @@ class Placements::AddSchoolContactWizard::SchoolContactStep < Placements::BaseSt
   attribute :last_name
   attribute :email_address
 
-  validates :first_name, :last_name, presence: true
   validates :email_address, presence: true, format: { with: URI::MailTo::EMAIL_REGEXP }
   validate :new_school_contact
 
   delegate :school, to: :wizard
 
   def new_school_contact
+    return if wizard.is_a?(Placements::EditSchoolContactWizard)
+
```

Comment:
I wonder if a more intention revealing name for this validation method would be `only_one_contact_per_school`?

Also, I think we could consolidate these two guard clauses into one (thanks to Rails 7's [excluding](https://blog.saeloun.com/2021/03/08/rails-6-1-adds-excluding-to-active-record-relation/) method):

```ruby
def only_one_contact_per_school
  another_contact = Placements::SchoolContact.where(school:).excluding(school_contact)
  return unless another_contact.exists?

  errors.add(:email_address, :taken)
end
```

This implementation should cover both scenarios – i.e. creating new contacts, and editing existing ones – because `excluding` works correctly with both `self.new_record?` and `self.persisted?` objects.

It also means the coupling between this step class and its parent wizard(s) becomes looser. It feels a bit awkward for the step to change its validation behaviour based on the wizard's class name.

Reply:
If you _do_ choose to go with this approach, you may also want to memoize the `school_contact` method.

```ruby
def school_contact
  @school_contact ||= school.build_school_contact(first_name:, last_name:, email_address:)
end
```

Or actually, since we might be editing the existing one...

```ruby
def school_contact
  return wizard.school_contact if wizard.respond_to?(:school_contact)

  @school_contact ||= school.build_school_contact(first_name:, last_name:, email_address:)
end
```